src/Services/Readers/Reader.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace LaravelEnso\DataImport\Services\Readers;
+
+use Box\Spout\Reader\IteratorInterface;
+use Exception;
+use LaravelEnso\DataImport\Exceptions\Import;
+
+abstract class Reader
```

Comment:
declare the abstract methods which should be implemented ( for example rowIterator() ).

maybe also create an interface which this abstract class should implement. you can also use that interface to typehint the reader property in Sheet.php. 

also use the Interface as a return type for methods ( ie reader($import)

Reply:
I also have a question, how do I resolve rowIterator in csv not using parameter and rowIterator in xlsx using parameter when it is called

Reply:
Also we have sheets() method not in csv

Reply:
one thing I believe you could do is implement a private $sheet property in XLSX.php

then, to set it on a specific instance you could make the sheet() method public and instead of returning `$iterator->current()` you should set the $sheet property. 

Example:

```
private function sheet(string $name): self
{
    ....
    $this->sheet = $iterator->current();
    return $this;
}
````

then you'll be able to use the service like so:

`$this->reader->sheet($this->sheet)->rowIterator();`

of course, you must also adjust the rowIterator() method. this is an idea from the top of my head, perhaps @aocneanu can also share his thoughts here.

Reply:
> Also we have sheets() method not in csv

this can be left "as is" in the XLSX.php service, doesn't need to be abstract or anything.

Reply:
Trying this make sheet also need a parameter for the two readers. CSV sheet doesn't need a sheet parameter.